homeassistant/components/niko_home_control/climate.py

```diff
@@ -0,0 +1,102 @@
+"""Support for Niko Home Control thermostats."""
+
+from typing import Any
+
+from nhc.const import THERMOSTAT_MODES
+from nhc.thermostat import NHCThermostat
+
+from homeassistant.components.climate import (
+    PRESET_ECO,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.components.sensor import UnitOfTemperature
+from homeassistant.const import ATTR_TEMPERATURE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import NikoHomeControlConfigEntry
+from .const import (
+    NIKO_HOME_CONTROL_THERMOSTAT_MODES,
+    NIKO_HOME_CONTROL_THERMOSTAT_MODES_MAP,
+)
+from .entity import NikoHomeControlEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: NikoHomeControlConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Niko Home Control thermostat entry."""
+    controller = entry.runtime_data
+
+    async_add_entities(
+        NikoHomeControlClimate(thermostat, controller, entry.entry_id)
+        for thermostat in controller.thermostats.values()
+    )
+
+
+class NikoHomeControlClimate(NikoHomeControlEntity, ClimateEntity):
+    """Representation of a Niko Home Control thermostat."""
+
+    _attr_supported_features: ClimateEntityFeature = (
+        ClimateEntityFeature.PRESET_MODE
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.TURN_OFF
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_name = None
+    _action: NHCThermostat
+
+    _attr_translation_key = "nhc_thermostat"
+
+    _attr_hvac_modes = [HVACMode.OFF, HVACMode.COOL, HVACMode.AUTO]
+
+    _attr_preset_modes = [
+        "day",
+        "night",
+        PRESET_ECO,
+        "prog1",
+        "prog2",
+        "prog3",
+    ]
+
+    def _get_niko_mode(self, mode: str) -> int:
+        """Return the Niko mode."""
+        for key, value in THERMOSTAT_MODES.items():
+            if value == mode:
+                return key
+        return NIKO_HOME_CONTROL_THERMOSTAT_MODES.OFF
```

Comment:
Why don't you just premake the THERMOSTAT_MODES to v,k

Reply:
need them both ways `off` -> `3` and `3` -> `off`

Reply:
```py
SWING_TO_FAN_OSCILLATION = {
    SWING_BOTH: "all",
    SWING_HORIZONTAL: "horizontal",
    SWING_VERTICAL: "vertical",
    SWING_OFF: "fixed",
}

FAN_OSCILLATION_TO_SWING = {
    value: key for key, value in SWING_TO_FAN_OSCILLATION.items()
}
```
example